김병대/4일차/boj-1068.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+using namespace std;
+
+class Node {
+    Node* parent = nullptr;
+    vector<Node*> childs;
+
+public:
+    // 부모 노드를 설정하고 부모 노드에게 자신을 자식으로 추가해달라고 한다.
+    void Init(Node* parent) {
+        this->parent = parent;
+        if(parent != nullptr) parent->AddChildNode(this);
+    }
```

Comment:
여기서 노드가 부모도 할당하게 코드를 작성하셨네요? 입력이 부모에 대한 정보가 오는 것은 맞지만, Leaf Node에서도 Root를 탐색할 수 있다면 이거는 트리보다는 그래프에 가깝지 않을까요?
```suggestion
    Node* parent = nullptr;
    vector<Node*> childs;

public:
    // 부모 노드를 설정하고 부모 노드에게 자신을 자식으로 추가해달라고 한다.
    void Init(Node* parent) {
        this->parent = parent;
        if(parent != nullptr) parent->AddChildNode(this);
    }
```